### PR TITLE
Feature - RadarChart Max value

### PR DESCRIPTION
**Pull request by DooRooSA**
Good day,

I refer you to: https://github.com/imaNNeoFighT/fl_chart/issues/958

This issue requested a max value for the RadarChart, which I have subsequently implemented.

(My first open source contribution, so please be gentle with feedback)

### CODE REVIEW

**Comment by imaNNeoFighT** [comment]
Hi. Well done.
I've never seen anyone make a great PR like this in the first contribution.
Everything is implemented in the correct way.

**Comment by imaNNeoFighT** [comment]
There is a minor conflict with CHANGELOG.md. 
Please fix it.

BTW you must write unit tests for the code you've written. (To keep the test coverage as high as possible. check out our [contribution gudeline](https://github.com/imaNNeoFighT/fl_chart/blob/master/CONTRIBUTING.md))

File: CHANGELOG.md [hunk 1]
```diff
@@ -413,4 +414,4 @@ LineTouchData(
 * restricted to access private classes of the library
 
 
-## 0.0.1 - Released on (2019 June 4)
\ No newline at end of file
+## 0.0.1 - Released on (2019 June 4)
```
Comment (imaNNeoFighT) [inline CHANGELOG.md:417]: Why this line has changed? Can you revert this change?

**Comment by FlorianArnould** [comment]
Is this PR still active ?
If not, I can refactor and add some tests to this.

**Comment by DooRooSA** [comment]
Good day,

I've got quite a rough deadline with my dayjob at the moment. I will be able to look at this again next weekend. If you would like to continue so long, please feel free to do so!

Have a great day!

**Comment by imaNNeoFighT** [comment]
@DooRooSA 
@FlorianArnould 
Can you please give an update about it? What did you do?

**Comment by FlorianArnould** [comment]
Nothing yet on my side, I worked more on the Gauge chart #199
I could have a look on this during the WE if needed

**Comment by imaNNeoFighT** [comment]
@DooRooSA what do you think?

**Comment by DooRooSA** [comment]
I will do the necessary changes right now, will let you know as soon as I'm done.

File: CHANGELOG.md (hunk 1 above)
Comment (DooRooSA) [reply CHANGELOG.md:417]: It seems that a newline character has been added to the end of this line, hence why it seems to have been edited. I'm busy figuring out how to undo that part of the commit

**Comment by DooRooSA** [comment]
Ok, I have created the unit tests for RadarChartHelper.

*Embarassing admission* This is the first time I have ever created a unit test. Please have a thorough look if I did it correctly.

File: CHANGELOG.md (hunk 1 above)
Comment (DooRooSA) [reply CHANGELOG.md]: Ok I tried to remove the newline character and committed it, but it seems it was unsuccessful. Are you able to change it from your side?

File: lib/src/chart/radar_chart/radar_chart_helper.dart [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import 'package:fl_chart/fl_chart.dart';
+import 'package:fl_chart/src/utils/list_wrapper.dart';
+
+/// Contains all the function to help the RadarChart work
+class RadarChartHelper {
+  /// Contains List of cached results, based on [List<RadarDataSet>]
+  ///
+  /// We use it to prevent redundant calculations
+  static final Map<ListWrapper<RadarDataSet>, RadarChartMinMaxAxisValues>
+      _cachedResults = {};
```
Comment (FlorianArnould) [inline lib/src/chart/radar_chart/radar_chart_helper.dart:11]: This cache is never cleaned.
This could lead to a memory leak

File: lib/src/chart/radar_chart/radar_chart_helper.dart [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+  static RadarChartMinMaxAxisValues calculateMinMaxAxisValue(
```
Comment (FlorianArnould) [inline lib/src/chart/radar_chart/radar_chart_helper.dart:13]: Why don't you just compute this in the `RadarChartData` contructor ?
with something like this : 
```dart
min = min ?? dataSets?.expand((s) => s.dataEntries).max.value ?? 0,
max = max ?? dataSets?.expand((s) => s.dataEntries).min.value ?? 0,
```
with this extension method or inline if you prefer:
```dart
extension DataSetLimits on Iterable<RadarEntry> {
  RadarEntry get max => reduce((a, b) => a.value > b.value ? a : b);
  RadarEntry get min => reduce((a, b) => a.value < b.value ? a : b);
}
```
I know that it generate one more iteration than your solution but it's a bit less complex and that's better than the previous implementation using a custom getter reevaluated generating 3 iterations per rendering

**Comment by zmeggyesi** [comment]
I believe #1091 implements this (along with defined minimum values)?

**Comment by imaNNeoFighT** [comment]
Yes. #1091 does the same thing we are going to implement here. Unfortunately, I will close this PR and we will continue with #1091

**Comment by imaNNeoFighT** [comment]
@FlorianArnould  and @ateich
I like your point of view in the PR review. May I ask you to review the #1091 branch too and tell me what you think?

File: lib/src/chart/radar_chart/radar_chart_helper.dart (hunk 2 above)
Comment (imaNNeoFighT) [reply lib/src/chart/radar_chart/radar_chart_helper.dart:11]: That's a great point. We have similar functionality like this in our LineChart and BarChart. 
The point is that as we recreate our chart data classes, we need to recalculate these values again and again. That's why I implemented this logic to handle a simple cache. And here, @DooRooSA followed what I did before.

File: lib/src/chart/radar_chart/radar_chart_helper.dart (hunk 3 above)
Comment (imaNNeoFighT) [reply lib/src/chart/radar_chart/radar_chart_helper.dart:13]: Check my answer [here](https://github.com/imaNNeoFighT/fl_chart/pull/995#discussion_r941729184)
